scheduler/replica_scheduler.go

```diff
@@ -0,0 +1,122 @@
+package scheduler
+
+import (
+	"github.com/rancher/longhorn-manager/datastore"
+	"github.com/rancher/longhorn-manager/types"
+	"github.com/rancher/longhorn-manager/util"
+
+	"fmt"
+	longhorn "github.com/rancher/longhorn-manager/k8s/pkg/apis/longhorn/v1alpha1"
+)
+
+const (
+	// longhornDirectory is the directory going to be bind mounted on the
+	// host to provide storage space to replica data by default
+	longhornDirectory = "/var/lib/rancher/longhorn/"
+)
+
+type ReplicaScheduler struct {
+	ds *datastore.DataStore
+}
+
+func NewReplicaScheduler(ds *datastore.DataStore) *ReplicaScheduler {
+	rcScheduler := &ReplicaScheduler{
+		ds: ds,
+	}
+	return rcScheduler
+}
+
+func (rcs *ReplicaScheduler) ScheduleReplica(replica *longhorn.Replica) (*longhorn.Replica, error) {
+	// only called when replica is starting for the first time
```

Comment:
If this function should only be called when `NodeID == ""`, then error out if NodeID is not empty. It will help to expose the potential bugs.